internal/step/plugin/provider.go

```diff
@@ -810,12 +941,25 @@ func (r *runningStep) run() {
 			r.logger.Warningf("failed to remove deployed container for step %s/%s", r.runID, r.pluginStepID)
 		}
 		r.lock.Unlock()
+		r.transitionToCancelled()
 		return
 	default:
 		r.container = container
 	}
 	r.lock.Unlock()
 	r.logger.Debugf("Successfully deployed container with ID '%s' for step %s/%s", container.ID(), r.runID, r.pluginStepID)
+
+	r.logger.Debugf("Checking to see if step %s/%s is enabled", r.runID, r.pluginStepID)
+	enabled, err := r.enableStage()
+	r.logger.Debugf("Step %s/%s enablement state: %t", r.runID, r.pluginStepID, enabled)
+	if err != nil {
+		// TODO
+		r.logger.Errorf("TODO: Error while checking stage enablement")
```

Comment:
This should be "todoed" before you're finished, shouldn't it?

Reply:
Replaced with a permanent solution.